JStanley_Oloughlin_Transmissivity_Calculation

```diff
@@ -0,0 +1,448 @@
+import math
+
+def baseflow_filter(y, a, BFImax):
+    """
+    Implementation of the Eckhardt 2005 (HP) recursive digital filter for baseflow separation. Courtesy of Dr. Ciaran Harman. 
+    """
+    N = len(y) # length of the timeseries
+    C = (1 - a) * BFImax / (1 - BFImax)
+    A = a / (1 + C)
+    B = C / (1 + C)
+    b = np.zeros(N)
+    b[0] = BFImax * np.mean(y)
+    for k in range(1,N):
+        if np.isnan(y[k]):
+            b[k] = A * b[k-1]
+        else:
+            b[k] = A * b[k-1] + B * y[k]
+        if b[k]>y[k]:
+            b[k] = y[k]
+    return b
+
+def mouse_event(event):
+    """
+    Parameters
+    ----------
+    event : mouse click
+       This is used so that index points can be found by clicking at any point on the graph.
+
+    Returns
+    -------
+    None.
+
+    """
+    print('x: {} and y: {}'.format(event.xdata, event.ydata))
+        
+def baseflow_calc(Q, P):
+    """
+    Parameters
+    ----------
+    Q : Pandas Dataframe
+        Column names 0, 1, 2, 3, 4, 5, 6. In order left to right: year, month, day, hour, minute, second, discharge (mm/day)
+    P : Pandas Dataframe
+        Column names 0, 1, 2, 3, 4, 5, 6. In order left to right: year, month, day, hour, minute, second, precipitation (mm/day)
+   
+    Returns
+    -------
+     baseflow : Arraay of float64
+        timeseries of baseflow (mm/day) in 15 minute segments in one column 
+
+    """
+    y = Q[6]
+    a = 0.9999
+    BFImax = 0.8
+    baseflow = baseflow_filter(y,a,BFImax)
+    return baseflow
+    
+def hydrograph(Q, P, baseflow):
+    """
+
+    Parameters
+    ----------
+    Q : Pandas Dataframe
+        Column names 0, 1, 2, 3, 4, 5, 6. In order left to right: year, month, day, hour, minute, second, discharge (mm/day)
+    P : Pandas Dataframe
+        Column names 0, 1, 2, 3, 4, 5, 6. In order left to right: year, month, day, hour, minute, second, precipitation (mm/day)
+    baseflow : Arraay of float64
+        timeseries of baseflow (mm/day) in 15 minute segments in one column 
+
+    Returns
+    -------
+    None.
+
+    """
+    x = Q.index
+    y1 = baseflow
+    y2 = Q[6]
+    y3 = P[6]
+    
+    fig, ax1 = plt.subplots(figsize=(12,10))
+    #Implementing mouse_event for the hydrograph
+    cid = fig.canvas.mpl_connect('button_press_event', mouse_event)
+    fig.subplots_adjust(bottom=0.3)
+    ax2 = ax1.twinx()
+    ax2.plot(x, y3, color = 'b')
+    ax2.set_ylabel(('Rainfall mm/hour'),
+                fontdict={'fontsize': 12})
+    # Invert y axis
+    ax2.invert_yaxis()
+    # Primary axes
+    ax1.plot(x, y1, color = 'r')#, linestyle='dashed', linewidth=1, markersize=12)
+    ax1.plot(x, y2, color = 'k')#, linestyle='dashed', linewidth=1, markersize=12)
+    # Define Labels
+    ax1.set_xlabel(('Date'),
+               fontdict={'fontsize': 14})
+    
+    ax1.set_ylabel(('Flow (mm/hour)'),
+                   fontdict={'fontsize': 14})
+    
+    ax1.set_title('Hydrograph Elder Creek', color = 'g')
```

Comment:
title set here by default - this should be an argument the function takes, or something it can find in the dataframe, if it is going to have a title at all.